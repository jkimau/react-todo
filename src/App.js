import React, { Component } from 'react';
import styled from 'styled-components';

const TopMenuBar = styled.div`
  height: 46px;
  border-bottom: 1px solid rgba(0,0,0,0.15);
`;

class App extends Component {
  render() {
    return (
      <div>
        <TopMenuBar>
          {/* <MenuBlock /> */}
          hello
        </TopMenuBar>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
