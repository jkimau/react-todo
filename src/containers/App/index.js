import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route
} from 'react-router-dom';

import AppContainer from 'containers/AppContainer';
import ListView from 'containers/ListView';
import CalendarView from 'containers/CalendarView';

class App extends Component {
  render() {
    return (
      <Router>
        <div>
          <Route path="/" component={AppContainer}/>
          <Route exact path="/" component={ListView}/>
          <Route path="/list" component={ListView}/>
          <Route path="/calendar" component={CalendarView}/>
        </div>
      </Router>
    )
  }
}

export default App;
