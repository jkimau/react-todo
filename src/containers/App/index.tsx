import React, { Component } from 'react';
import { withRouter } from 'react-router'
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { closeAllTodoMenus, setViewMode } from 'actions';
import TopNavigation from 'components/TopNavigation';
import SearchInput from 'components/Search';
import ListView from 'containers/ListView';
import CalendarView from 'containers/CalendarView';
import { lightGreyBorder } from 'global/colors';
import { topMenuHeight } from 'global/variables';

const TopMenuBar = styled.div`
  display: flex;
  height: ${topMenuHeight};
  border-bottom: 1px solid ${lightGreyBorder};
`;

const MainTitle = styled.div`
  flex-grow: 1;
  display: flex;
  align-items: center;
  justify-content: center;
  height: ${topMenuHeight};
`;

const SearchInputWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding-right: 20px;
  width: 33%;
  height: ${topMenuHeight};
`;

export interface AppProps {
  location: {
    pathname: string
  },
  viewMode: string,
  setViewMode(mode: string): void,
  closeAllTodoMenus(): void
}

export class App extends Component<AppProps, undefined> {
  constructor(props) {
    super(props);

    this.documentClickHandler = this.documentClickHandler.bind(this);
    this.changeViewModeHandler = this.changeViewModeHandler.bind(this);
    this.setInitialViewMode = this.setInitialViewMode.bind(this);

    this.setInitialViewMode();
  }

  changeViewModeHandler(mode: string) {
    this.props.setViewMode(mode)
  }

  documentClickHandler(e) {
    const classNames = e.target.className.split(' ');
    if (!classNames.includes('todo-menu-trigger')) {
      this.props.closeAllTodoMenus();
    }
  }

  setInitialViewMode() {
    const path = this.props.location.pathname;
    const mode = path.replace(/\//, '');
    this.props.setViewMode(mode)
  }

  render() {
    return (
      <div className="app-container" onClick={this.documentClickHandler}>
        <TopMenuBar>
          <TopNavigation
            viewMode={this.props.viewMode}
            changeViewModeHandler={this.changeViewModeHandler}
          />
          <MainTitle>TODO</MainTitle>
          <SearchInputWrapper>
            <SearchInput />
          </SearchInputWrapper>
        </TopMenuBar>

        <Route exact path="/" component={ListView}/>
        <Route path="/list" component={ListView}/>
        <Route path="/calendar" component={CalendarView}/>
      </div>
    )
  }
}

const mapStateToProps = ({ todoState: { viewMode }}) => {
  return { viewMode }
};

const mapDispatchToProps = dispatch => {
  return {
    closeAllTodoMenus: () => { dispatch(closeAllTodoMenus()) },
    setViewMode: (mode) => { dispatch(setViewMode(mode)) }
  }
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(App)
);
