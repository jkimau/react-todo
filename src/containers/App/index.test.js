import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

// import TopNavigation from 'components/TopNavigation';
// import SearchInput from 'components/Search';
// import ListView from 'containers/ListView';
// import CalendarView from 'containers/CalendarView';
import { App } from '../App';

describe('<App/>', () => {
  const defaultProps = {
    viewMode: 'list',
    closeAllTodoMenus: jest.fn(),
    setViewMode: jest.fn(),
    location: {
      pathname: 'fake_pathname'
    }
  };

  const compileComponent = (props = defaultProps) => {
    return shallow(<App {...props}/>)
  };

  // test('snapshot with view mode list', () => {
  //   const tree = renderer.create(
  //     <MemoryRouter>
  //       <App {...defaultProps}/>
  //     </MemoryRouter>
  //   ).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  test('should call setInitialViewMode() on load', () => {
    App.prototype.setInitialViewMode = jest.fn();
    compileComponent();

    expect(App.prototype.setInitialViewMode).toHaveBeenCalled();
  });

  test('should render <Route/> properly', () => {
    const wrapper = compileComponent();

    expect(wrapper.find('Route').length).toBe(3);
  });

  test('should pass correct props to <TopNavigation/>', () => {
    const wrapper = compileComponent();

    expect(wrapper.find('TopNavigation').props()).toEqual({
      viewMode: 'list',
      changeViewModeHandler: wrapper.instance().changeViewModeHandler
    })
  });
});
