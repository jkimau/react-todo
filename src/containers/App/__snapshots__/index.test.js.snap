exports[`<App/> snapshot - list view mode 1`] = `
<div
  className="app-container"
  onClick={[Function]}>
  <div
    className="jfwhKk">
    <nav
      className="fBEMhB">
      <ul>
        <li
          className="gxaXoS">
          <a
            href="/list"
            id="list-nav-link"
            onClick={[Function]}
            style={
              Object {
                "color": "blue",
                "textDecoration": "none",
              }
            }>
            List view
          </a>
        </li>
        <li
          className="gxaXoS">
          <a
            href="/calendar"
            id="calendar-nav-link"
            onClick={[Function]}
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }>
            Calendar view
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="bYIAja">
      TODO
    </div>
    <div
      className="dXtVax">
      <input
        className="kTPKMA"
        placeholder="Please enter keywords"
        type="text" />
    </div>
  </div>
</div>
`;

exports[`<App/> snapshot - list view mode 2`] = `
<div
  className="app-container"
  onClick={[Function]}>
  <div
    className="jfwhKk">
    <nav
      className="fBEMhB">
      <ul>
        <li
          className="gxaXoS">
          <a
            href="/list"
            id="list-nav-link"
            onClick={[Function]}
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }>
            List view
          </a>
        </li>
        <li
          className="gxaXoS">
          <a
            href="/calendar"
            id="calendar-nav-link"
            onClick={[Function]}
            style={
              Object {
                "color": "blue",
                "textDecoration": "none",
              }
            }>
            Calendar view
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="bYIAja">
      TODO
    </div>
    <div
      className="dXtVax">
      <input
        className="kTPKMA"
        placeholder="Please enter keywords"
        type="text" />
    </div>
  </div>
</div>
`;
